### ISSUE TITLE
Cloud synchronization

### ISSUE THREAD

**Issue by tinkicker** [federicoiosue/Omni-Notes#55]

**Comment by pejakm** [comment]
A synchronization to ownCloud Notes would be great to have!

**Comment by federicoiosue** [comment]
Yes, it would! But not being an ownCloud user I think I'll not develop it personally. 

I'll e glad to evaluate pull requests anyhow!

**Comment by faflimadafakaz** [comment]
Will Cloud sync be something made soon?

**Comment by lafener** [comment]
I wonder using APIs provided by Evernote to achieve sync would be a good choice.

**Comment by faflimadafakaz** [comment]
Something new here?

**Comment by samever7** [comment]
I am waiting for it ~~

**Comment by Rainer-Lang** [comment]
That would be a really great feature.

**Comment by gnacho** [comment]
Is this feature going to be developed? Should be just great having such an amazing app notes synchronized with open source plattform as well :+1:

**Comment by r0bbie** [comment]
Also really keen to see a sync feature! Really liking the Omni Notes app, and this is the only thing that's holding back it's usefulness day-to-day.

As another possible synchronisation option, I'd put forward [remoteStorage](https://remotestorage.io), which could allow syncing a number of open source desktop note-taking apps, such as [Laverna](https://laverna.cc/).

**Comment by federicoiosue** [comment]
Very interesting option @gnacho, I'll give a look at that, thanks

**Comment by supportreq** [comment]
@federicoiosue is this in works??

**Comment by federicoiosue** [comment]
Stay tuned with this branch https://github.com/federicoiosue/Omni-Notes

**Comment by federicoiosue** [comment]
Now that I'm hardening both the Android app and the desktop application I'm back on the cloud-sync topic.

Due to the fact that I've an "unlimited" PHP hostin plan, after evaluating some nice Java-based solution (mainly BaaS) I've now found an interesting solution on [Hook](http://hook-platform.org/#/features).

Someone has experience with it? @r0bbie (are you the guy from F-Droid?)?

**Comment by rugk** [comment]
NextCloud sync (like https://github.com/stefan-niedermann/nextcloud-notes does) would be great.

**Comment by andre-dierker** [comment]
Maybe a WebDAV solution is widelyer usable. NextCloud, Homeserver, bare Webserver and many more.

You could take a look at
https://github.com/yeonsh/Sardine-Android

**Comment by federicoiosue** [comment]
I've prepared a Nextcloud instance to start working with their SDK, it seems an interesting solution that could be implemented both as a centralized cloud service or a distribuited home-service.

**Comment by rugk** [comment]
Maybe better start with a centralized service… Most NextCloud users self-host their server and have all their data there so that's okay.
Of course, you should allow users to enter their own NextCloud URL…

**Comment by federicoiosue** [comment]
That's the plan @rugk 

It should be noted that keeping central instance of Nextcloud publically accessible will take me a certain amount of time

**Comment by rugk** [comment]
Ah, so you want to offer a NextCloud instance for this notes app?
I think this is not necessary as there are some new users can use and most NextCloud users already use a self hosted version.
So I would prioritise the implementation work. All other NextCloud apps I have seen also do not have a default server.

**Comment by gnacho** [comment]
I left Omninotes cause its lack of cloud sync (luckily cause I lost two phones in this time), but now I'm back trying the app and it's exactly what I need/want. But sad to see that doesn't have any sync yet.
Any plans for do it soon?

**Comment by anarchist912** [comment]
@federicoiosue Are you still working on the Nextcloud sync functionality? It would be amazing to have!!
In the meantime, is there a way to export and import manually? E.g. in order to elaborate on the notes on my laptop using copy&paste?

**Comment by kpuljek** [comment]
I would love to be able to use Omni-Notes through my NextCloud, +1

**Comment by blubbsy** [comment]
me too +1

**Comment by jaydeland** [comment]
Looking for this also

**Comment by mweisshaupt1988** [comment]
This is the only thing that is keeping me from using Omni-Notes. I don't have much free time to help with development currently but I would pay a few € to someone that implemented Nextcloud Notes sync ;-)

**Comment by kpuljek** [comment]
NC notes aren't really feature rich, they are just .txt files. Maybe Omni syncing should be implemented as a separate thing and not using the NextCloud notes

**Comment by rugk** [comment]
Maybe it can just save them as text files (some markdown format or whatever) and let Nextcloud sync it. (when https://github.com/nextcloud/android/issues/285 is implemented)

**Comment by jnko** [comment]
@federicoiosue Are you still working on this feature? It's have been about 4 years and three phones ago and there's still no other notes takeing app I trust. But that Sync-Thing would be really fine to have. At least some kind of dumb device-sync to webdav (probably SyncML?) would be enough . Editing via browser would be nice but too far out of focus I believe...

**Comment by seth100** [comment]
+1 for Nextcloud sync

**Comment by philiplb** [comment]
WebDAV-Sync would be so nice. And would cover Nextcloud and many others (OwnCloud, HiDrive, ...)

**Comment by x11x** [comment]
Just wanted to mention the issue I opened #534 suggesting [Sandstorm.io](https://sandstorm.io) as a platform for an open source, self-hostable cloud service

**Comment by gjelsas** [comment]
CalDAV +1

**Comment by erduende** [comment]
Is there any news about Omni Notes cloud synchronization?

**Comment by bpcurse** [comment]
Joplin now has a nextcloud API (+ E2EE). Due to that I have started using it in parallel.
Omni-Notes is beautiful and has some advantages, but in the long term cloud sync is essential and with the massive propagation of private nextcloud servers will probably make the difference for many users.

**Comment by federicoiosue** [comment]
> Joplin now has a nextcloud API (+ E2EE). Due to that I have started using it in parallel.
> Omni-Notes is beautiful and has some advantages, but in the long term cloud sync is essential and with the massive propagation of private nextcloud servers will probably make the difference for many users.

That's a realy great app, with a good team behind it and they've been capable to build a really fantastic ecosystem around that. But talking about me, as a single developer, focusing on specific APIs , expecially if not adoptable by a large number of users, is not possible.

What's the status of the sync feature: I've recently worked on code cleaning and making it work again also for F-Droid builds, added tests to make it more robust, released a couple of big features (JSON backup first of all) that are part of the sync processing. 

More infos are available for everyone on the [project board](https://github.com/federicoiosue/Omni-Notes/projects/1).

**Comment by keunes** [comment]
Just to chip inwirth my +1: I really likeOmniNotes. I've tried Joplin but it's by far not as friendly, and now I'm usingthre semi-official Nextcloud Notes app (fine but still prefer ON).
On a related note: how are notes stored by ON? Asking because if saved as MD files with attachments, and if we can set the storage folder, we might just be able to sync using the Nextcloud app (and edit them online & on Desktop via respective apps).

**Comment by federicoiosue** [comment]
@keunes data are stored into a database, so they can't be synchronized. 

Starting from version 6.x.x anyway backups are available in JSON format into the folder "Omni Notes" of your internal device memory, to allow compatibility with the format used by [the desktop app](https://github.com/federicoiosue/omni-notes-desktop). While you already could sync them with the cloud (I do it myself to keep them safe) there is a currently active branch of development that aims to [perform automatically](https://github.com/federicoiosue/Omni-Notes/blob/feature/autobackup/omniNotes/src/main/res/xml/settings_sync_import_export.xml#L39) this thing.

As stated above that feature is currently under development but, due to the fact that I'm dealing with users data I really want to be sure to have the underlying code as much stable and robust as possibile, and that's what the newcoming version [6.1.0](https://github.com/federicoiosue/Omni-Notes/issues?q=is%3Aopen+is%3Aissue+milestone%3A6.1.0) is is aiming to.

**Comment by nacho-domatix** [comment]
> Just to chip inwirth my +1: I really likeOmniNotes. I've tried Joplin but it's by far not as friendly, and now I'm usingthre semi-official Nextcloud Notes app (fine but still prefer ON).
> On a related note: how are notes stored by ON? Asking because if saved as MD files with attachments, and if we can set the storage folder, we might just be able to sync using the Nextcloud app (and edit them online & on Desktop via respective apps).

Nextcloud notes does the job pretty good, at least for me :)
